source/draw/Image.ooc

```diff
@@ -38,22 +31,39 @@ Image: abstract class {
 		if (this referenceCount != null)
 			this referenceCount free()
 		this _referenceCount = null
-		if (this _canvas != null)
-			this _canvas free()
-		this _canvas = null
 		super()
 	}
-	drawPoint: func ~white (position: FloatPoint2D) { this canvas drawPoint~white(position) }
-	drawPoint: func ~explicit (position: FloatPoint2D, pen: Pen) { this canvas drawPoint~explicit(position, pen) }
-	drawLine: func ~white (start, end: FloatPoint2D) { this canvas drawLine~white(start, end) }
-	drawLine: func ~explicit (start, end: FloatPoint2D, pen: Pen) { this canvas drawLine~explicit(start, end, pen) }
-	drawPoints: func ~white (pointList: VectorList<FloatPoint2D>) { this canvas drawPoints~white(pointList) }
-	drawPoints: func ~explicit (pointList: VectorList<FloatPoint2D>, pen: Pen) { this canvas drawPoints~explicit(pointList, Pen new(ColorRgba white)) }
-	drawLines: func ~white (pointList: VectorList<FloatPoint2D>) { this canvas drawLines~white(pointList) }
-	drawLines: func ~explicit (pointList: VectorList<FloatPoint2D>, pen: Pen) { this canvas drawLines~explicit(pointList, Pen new(ColorRgba white)) }
-	drawBox: func ~white (box: FloatBox2D) { this canvas drawBox~white(box) }
-	drawBox: func ~explicit (box: FloatBox2D, pen: Pen) { this canvas drawBox~explicit(box, Pen new(ColorRgba white)) }
-	fill: func ~implicitCanvas (color: ColorRgba) { this canvas fill(color) }
+	drawPoint: virtual func ~white (position: FloatPoint2D) { this drawPoint(position, Pen new(ColorRgba white)) }
+	drawPoint: virtual func ~explicit (position: FloatPoint2D, pen: Pen) {
+		list := VectorList<FloatPoint2D> new()
+		list add(position)
+		this drawPoints(list, pen)
+		list free()
+	}
+	drawLine: virtual func ~white (start, end: FloatPoint2D) { this drawLine(start, end, Pen new(ColorRgba white)) }
+	drawLine: virtual func ~explicit (start, end: FloatPoint2D, pen: Pen) {
```

Comment:
Can't you do `pen := Pen new(ColorRgba White)` in the arguments and avoid the extra declaration above? Same with many methods here.